exec.go

```diff
@@ -78,85 +73,49 @@ func ExecCommand(ui Ui, input ExecCommandInput) {
 	if writeEnv {
 		ui.Debug.Println("Writing temporary credentials to ENV")
 
-		env = overwriteEnv(env, "AWS_ACCESS_KEY_ID", val.AccessKeyID)
-		env = overwriteEnv(env, "AWS_SECRET_ACCESS_KEY", val.SecretAccessKey)
+		env.Set("AWS_ACCESS_KEY_ID", val.AccessKeyID)
+		env.Set("AWS_SECRET_ACCESS_KEY", val.SecretAccessKey)
 
 		if val.SessionToken != "" {
-			env = overwriteEnv(env, "AWS_SESSION_TOKEN", val.SessionToken)
-			env = overwriteEnv(env, "AWS_SECURITY_TOKEN", val.SessionToken)
-		}
-	}
-
-	cmd := exec.Command(input.Command, input.Args...)
-	cmd.Env = env
-	cmd.Stdin = os.Stdin
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
-
-	go func() {
-		sig := <-input.Signals
-		if cmd.Process != nil {
-			cmd.Process.Signal(sig)
-		}
-	}()
-
-	var waitStatus syscall.WaitStatus
-	if err := cmd.Run(); err != nil {
-		if err != nil {
-			ui.Error.Println(err)
-		}
-		if exitError, ok := err.(*exec.ExitError); ok {
-			waitStatus = exitError.Sys().(syscall.WaitStatus)
-			os.Exit(waitStatus.ExitStatus())
+			env.Set("AWS_SESSION_TOKEN", val.SessionToken)
+			env.Set("AWS_SECURITY_TOKEN", val.SessionToken)
 		}
 	}
-}
 
-// write out a config excluding role switching keys
-func writeTempConfig(profile string, conf profiles) (*os.File, error) {
-	tmpConfig, err := ioutil.TempFile(os.TempDir(), "aws-vault")
+	path, err := exec.LookPath(input.Command)
 	if err != nil {
-		return nil, err
+		ui.Error.Fatal(err)
 	}
 
-	newConfig := map[string]string{}
+	pid, err := syscall.ForkExec(path, append([]string{input.Command}, input.Args...), &syscall.ProcAttr{
+		Env:   env,
+		Files: []uintptr{0, 1, 2},
+	})
```

Comment:
I think `syscall.ForkExec()` has most of the same issues that https://github.com/99designs/aws-vault/issues/44 describes (i.e. it's still a subprocess), and that `syscall.Exec()` would be required to go back to single-process simplicity. That wont be viable in server mode, but perhaps in env mode?